worlds/mlss/Rules.py

```diff
@@ -0,0 +1,307 @@
+        add_rule(world.multiworld.get_location("Chucklehuck Woods Pipe 5 Room Coin Block", world.player),
+                 lambda state: StateLogic.brooch(state, world.player) and StateLogic.hammers(state, world.player))
+        add_rule(world.multiworld.get_location("Chucklehuck Woods Room 7 Coin Block", world.player),
+                 lambda state: StateLogic.brooch(state, world.player) and StateLogic.hammers(state, world.player))
+        add_rule(world.multiworld.get_location("Chucklehuck Woods After Chuckleroot Coin Block", world.player),
+                 lambda state: StateLogic.brooch(state, world.player) and StateLogic.fruits(state, world.player))
+        add_rule(world.multiworld.get_location("Chucklehuck Woods Koopa Room Coin Block", world.player),
+                 lambda state: StateLogic.brooch(state, world.player))
+        add_rule(world.multiworld.get_location("Chucklehuck Woods Winkle Area Cave Coin Block", world.player),
+                 lambda state: StateLogic.brooch(state, world.player) and StateLogic.canDash(state, world.player))
+        add_rule(world.multiworld.get_location("Sewers Prison Room Coin Block", world.player),
+                 lambda state: StateLogic.rose(state, world.player))
+        add_rule(world.multiworld.get_location("Teehee Valley Past Ultra Hammer Rocks Coin Block", world.player),
+                 lambda state: StateLogic.ultra(state, world.player))
+        add_rule(world.multiworld.get_location("S.S. Chuckola Storage Room Coin Block 1", world.player),
+                 lambda state: StateLogic.super(state, world.player) or StateLogic.canDash(state, world.player))
+        add_rule(world.multiworld.get_location("Gwarhar Lagoon First Underwater Area Room 2 Coin Block", world.player),
+                 lambda state: StateLogic.canDash(state, world.player) and (StateLogic.membership(state, world.player) or StateLogic.surfable(state, world.player)))
+        add_rule(world.multiworld.get_location("S.S. Chuckola Storage Room Coin Block 2", world.player),
+                 lambda state: StateLogic.super(state, world.player) or StateLogic.canDash(state, world.player))
+        add_rule(world.multiworld.get_location("Joke's End Second Floor West Room Coin Block", world.player),
+                 lambda state: StateLogic.ultra(state, world.player) and StateLogic.fire(state, world.player)
+                 and (StateLogic.membership(state, world.player)
+                 or (StateLogic.canDig(state, world.player) and StateLogic.canMini(state, world.player))))
+        add_rule(world.multiworld.get_location("Joke's End North of Bridge Room Coin Block", world.player),
+                 lambda state: StateLogic.ultra(state, world.player) and StateLogic.fire(state, world.player)
+                 and StateLogic.canDig(state, world.player)
+                 and (StateLogic.membership(state, world.player)
+                 or StateLogic.canMini(state, world.player)))
```

Comment:
For readability, I think that an `or` chain inside an overarching `and` chain should be indented

```
StateLogic.ultra(state, world.player) 
and StateLogic.fire(state, world.player)
and StateLogic.canDig(state, world.player)
and (
    StateLogic.membership(state, world.player)
    or StateLogic.canMini(state, world.player)
)
```

*Low priority*